test/integration/shoots/applications/shoot_app_test.go

```diff
@@ -0,0 +1,332 @@
+		Expect(err).NotTo(HaveOccurred())
+
+		redisSlaveLabels := map[string]string{
+			"app":  "redis",
+			"role": "slave",
+		}
+
+		err = shootTestOperations.WaitUntilPodWithLabelsIsReady(guestBookAppCtx, redisSlaveLabels, helmDeployNamespace, shootTestOperations.K8sShootClient, 5*time.Second)
+		Expect(err).NotTo(HaveOccurred())
+
+		guestBookParams := struct {
+			HelmDeployNamespace string
+			ShootDNSHost        string
+		}{
+			helmDeployNamespace,
+			fmt.Sprintf("guestbook.ingress.%s", *shootTestOperations.Shoot.Info.Spec.DNS.Domain),
+		}
+
+		// Parse the guestbook app template
+		var writer bytes.Buffer
+		err = guestBooktpl.Execute(&writer, guestBookParams)
+		Expect(err).NotTo(HaveOccurred())
+
+		// Apply the guestbook app resources to shoot
+
+		manifestReader := kubernetes.NewManifestReader(writer.Bytes())
+		err = shootTestOperations.K8sShootClient.Applier().ApplyManifest(context.TODO(), manifestReader)
+		Expect(err).NotTo(HaveOccurred())
+
+		// define guestbook app urls
+		guestBookAppURL := fmt.Sprintf("http://guestbook.ingress.%s", *shootTestOperations.Shoot.Info.Spec.DNS.Domain)
+		pushUrl := fmt.Sprintf("%s/rpush/guestbook/foobar-%s", guestBookAppURL, shootTestOperations.Shoot.Info.Name)
+		pullUrl := fmt.Sprintf("%s/lrange/guestbook", guestBookAppURL)
+
+		// Check availability fo the guestbookapp
+		err = shootTestOperations.WaitUntilGuestbookAppIsAvailable(guestBookAppCtx, []string{guestBookAppURL, pushUrl, pullUrl}, 10*time.Second)
+		Expect(err).NotTo(HaveOccurred())
+
+		// Push foobar-<shoot-name> to the guestbook app
+		_, err = shootTestOperations.HTTPGet(guestBookAppCtx, pushUrl)
+		Expect(err).NotTo(HaveOccurred())
+
+		// Pull foobar
+		pullResponse, err := shootTestOperations.HTTPGet(guestBookAppCtx, pullUrl)
+		Expect(err).NotTo(HaveOccurred())
+
+		responseBytes, err := ioutil.ReadAll(pullResponse.Body)
+		Expect(err).NotTo(HaveOccurred())
+
+		// test if foobar-<shoot-name> was pulled successfully
+		bodyString := string(responseBytes)
+		Expect(bodyString).To(ContainSubstring(fmt.Sprintf("foobar-%s", shootTestOperations.Shoot.Info.Name)))
+
+		shootTestOperations.ShootGardenerTest.Logger.Info("Guestbook app was deployed successfully!")
+
+	})
+
+	It("Dashboard should be available", func() {
+		ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+		defer cancel()
+
+		err := shootTestOperations.DashboardAvailable(ctx)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	Context("Network Policy Testing", func() {
+		var (
+			// context is implemented at the moment for the generated clients so we use a TODO
+			ctx      = context.TODO()
+			testFunc = func(host, port string, expected int) {
+				err := shootTestOperations.PodExecByLabel(ctx, apiserverLabels, APISERVER,
+					fmt.Sprintf("nc -z -w5 %s %s", host, port), framework.Seed)
+				switch expected {
```

Comment:
Instead of having a `switch` here that differentiates two test cases, create two 'test tables' with different assertions, then you can group your tests nicely and also your `It`s will have proper descriptions.